keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -238,4 +315,41 @@ class SqlVerificationRepository(
       .from(DELIVERY_ARTIFACT)
       .where(DELIVERY_ARTIFACT.DELIVERY_CONFIG_NAME.eq(deliveryConfig.name))
       .and(DELIVERY_ARTIFACT.REFERENCE.eq(artifactReference))
+
+
+  /**
+   * Helper class for [getStatesBatch]
+   */
+  @Suppress("PropertyName")
+  private class ContextTable(
+    val contexts: List<VerificationContext>,
+    val jooq: DSLContext
+  ) {
+    val alias = "verification_contexts"
+
+    private val ind = "ind"
+    private val environmentName = "environment_name"
+    private val artifactReference = "artifact_reference"
+    private val artifactVersion = "artifact_version"
+
+    fun <T> f(s : String, t: Class<T>) : Field<T> = field(name(alias, s), t)
+
+    // These behave like regular jOOQ table field names when building SQL queries
+
+    val IND  = f(ind, Long::class.java)
+    val ENVIRONMENT_NAME = f(environmentName, String::class.java)
+    val ARTIFACT_REFERENCE = f(artifactReference, String::class.java)
+    val ARTIFACT_VERSION = f(artifactVersion, String::class.java)
+
+    /**
+     * return a jOOQ table that contains the [contexts] data represented as a table that can be selected against
+     */
+    val table : Table<Record4<Int, String, String, String>>
+      get() =
+        contexts
+          .mapIndexed { idx, v -> jooq.select(inline(idx), inline(v.environmentName), inline(v.artifactReference), inline(v.version)) as SelectOrderByStep<Record4<Int, String, String, String>> }
+          .reduce { s1, s2 -> s1.unionAll(s2) }
+          .asTable(alias, ind, environmentName, artifactReference, artifactVersion)
```

Comment:
I can't say I fully understand this block of code. Do you think it might be worth expanding the comment above to add a bit more detail about what this does, especially the map/reduce part?